.github/workflows/safari_stable.yml

```diff
@@ -17,8 +22,21 @@ env:
   SAFARIDRIVER_DIAGNOSE: false
 
 jobs:
+  check-workflow-run:
+    name: "Check for appropriate epochs"
+    uses: ./.github/workflows/check-workflow-run.yml
+    with:
+      check-refs: '["refs/heads/epochs/daily"]'
+    permissions:
+      actions: read
+
   safari-stable-results:
     name: "All Tests: Safari (stable)"
+    needs: check-workflow-run
+    if: |
+      always() &&
```

Comment:
Could we have a comment here? I don't know why `always()` is needed. The first condition after that makes sense, and I guess the second is "run if we were manually triggered, or if there's an updated ref", but it would be nice to be sure.

Reply:
Not just manually triggered, also a push to `epochs/three_hourly` or `triggers/safari_stable` that hasn't come from a GitHub Actions GitHub token (though obviously there's unlikely to be any of the former).